jwst/photom/photom.py

```diff
@@ -1007,13 +1029,50 @@ def create_2d_conversion(self, model, exptype, conversion, waves, relresps,
         # 2D wavelength grid
         conv_2d = np.interp(wl_array, waves, relresps, left=np.nan, right=np.nan)
 
+        if include_dispersion:
+            dispaxis = get_dispersion_direction(self.exptype, self.grating, self.filter, self.pupil)
+            if dispaxis is not None:
+                dispersion_array = self.get_dispersion_array(wl_array, dispaxis)
+                # Convert dispersion from micron/pixel to angstrom/pixel
+                dispersion_array *= 1.0e4
+                conv_2d /= np.abs(dispersion_array)
+            else:
+                log.warning("Unable to get dispersion direction, so cannot calculate dispersion array")
         # Combine the scalar and 2D conversion factors
         conversion = conversion * conv_2d
         no_cal = np.isnan(conv_2d)
         conversion[no_cal] = 0.
 
         return conversion, no_cal
 
+    def get_dispersion_array(self, wavelength_array, dispaxis):
+        """Create an array of dispersion values from the wavelength array
+
+        Parameters
+        ----------
+        wavelength_array : float
+            2-d array of wavelength values, assumed to be in microns
+        dispaxis : integer
+            Direction of dispersion: 1 = along columns, 2 = along rows
+
+        Returns
+        -------
+        dispersion_array : float
+            2-d array of dispersion values, in microns/pixel
+
+        """
+        nrows, ncols = wavelength_array.shape
+        dispersion_array = np.zeros(wavelength_array.shape)
+        if dispaxis == 1:
+            for row in range(nrows):
+                dispersion_array[row] = np.gradient(wavelength_array[row])
+        elif dispaxis == 2:
+            for column in range(ncols):
+                dispersion_array[:, column] = np.gradient(wavelength_array[:, column])
+        else:
+            log.warning(f"Can't process data with DISPAXIS={dispaxis}")
```

Comment:
5 is right out!  ;-)

Reply:
A very short fuse on the DISPAXIS hand grenade...